Sources/CSVDealer/CSVDealer.swift

```diff
@@ -54,12 +54,18 @@ public struct CSVParser<T: CSVConvertible> {
         return (headers, nonHeaderRows)
     }
     
-    public func parse() throws -> [T] {
+    public func parse(lineCountToParse: Int? = nil) throws -> [T] {
         let decoder = CSVDecoder<T>()
         var results: [T] = []
         
+        let rowsToParse:[[Any]]
+        if let lineLimit = lineCountToParse {
+            rowsToParse = Array(nonHeaderRows.prefix(lineLimit))
+        }else {
+            rowsToParse = nonHeaderRows
+        }
```

Comment:
Looks good. One important point though. The idea for this is that it returns a random set of rows. That way we can create a unique experience for users every time (most times) they open a lesson.

<img width="200" alt="Screenshot 2025-03-26 at 12 08 30 AM" src="https://github.com/user-attachments/assets/9b6c4d92-cde3-41bc-a003-721f8eedbe6f" />

```suggestion
        let rowsToParse:[[Any]]
        if let lineLimit = lineCountToParse {
            rowsToParse = Array(nonHeaderRows.shuffled().prefix(lineLimit))
        } else {
            rowsToParse = nonHeaderRows
        }
```

Reply:
You are right. so i have changed code in another branches and made PR